packages/libraries/linkable-specs/src/__tests__/link.spec.ts

```diff
@@ -0,0 +1,72 @@
+import { parse } from 'graphql';
+import { FederatedLink } from '../link';
+
+function trimMultiline(str: string): string {
+  return str
+    .split('\n')
+    .map(s => s.trim())
+    .filter(l => !!l)
+    .join('\n');
+}
+
+describe('FederatedLink', () => {
+  test.each([
+    `
+      extend schema
+        @link(url: "https://specs.apollo.dev/link/v1.0")
+        @link(url: "https://specs.apollo.dev/federation/v2.0")
+    `,
+    `
+      extend schema
+        @link(url: "https://specs.apollo.dev/link/v1.0", as: "@linkz")
+        @link(url: "https://specs.apollo.dev/federation/v2.0", as: "fed", import: ["@key"])
+    `,
+    `
+      extend schema
+        @link(url: "https://specs.apollo.dev/link/v1.0")
+        @link(url: "https://specs.apollo.dev/federation/v2.0", import: [{ name: "@key", as: "@lookup" }])
+    `,
+  ])('fromTypedefs', (sdl: string) => {
+    // string manipulate to extract just the link trings
+    const firstLinkPos = sdl.indexOf('@link(');
+    const linksOnly = trimMultiline(sdl.substring(firstLinkPos));
+    // compare to parsed result
+    const typeDefs = parse(sdl);
+    const links = FederatedLink.fromTypedefs(typeDefs);
+    expect(links.join('\n')).toBe(linksOnly);
+  });
+
+  test('resolveImportName', () => {
+    const sdl = `
+      extend schema
+        @link(url: "https://specs.apollo.dev/link/v1.0")
+        @link(url: "https://specs.apollo.dev/federation/v2.0", import: [{ name: "@key", as: "@lookup" }, "@provides"])
+        @link(url: "https://unnamed.graphql-hive.com/v0.1", import: ["@meta"])
+        @link(url: "https://specs.graphql-hive.com/hive/v0.1", import: ["@group"], as: "hivelink")
+    `;
+    const links = FederatedLink.fromTypedefs(parse(sdl));
+    const federationLink = links.find(l => l.identity === 'https://specs.apollo.dev/federation');
+    expect(federationLink).toBeDefined();
+    // aliased
+    expect(federationLink?.resolveImportName('@key')).toBe('lookup');
```

Comment:
The `@` is intentionally stripped to make it easier to use. All implementations I've seen for visiting/mapping schema use the name of the nodes, and a directive's name doesn't include the `@`. However, the `@` is necessary for the parameter so that I can know how to resolve the name. Otherwise, setting the import argument as: `import: ["foo"]` would affect both `@foo` and `foo`.

Reply:
Added to the `resolveImportName` function jsdoc